core/os/flock/flock.go

```diff
@@ -18,112 +18,134 @@ import (
 	"errors"
 	"os"
 	"path/filepath"
-	"strings"
 	"sync"
 	"time"
 )
 
```

Comment:
This is all much cleaner, thank you.
I don't see any sort of timeout on these locks. If a process crashes with the lock held, how do we recover?